airbyte-integrations/connectors/source-freshdesk/source_freshdesk/source.py

```diff
@@ -52,53 +51,50 @@ def __init__(self, api_key: str) -> None:
 
 
 class SourceFreshdesk(AbstractSource):
+    @staticmethod
+    def _get_stream_kwargs(config: Mapping[str, Any]) -> dict:
+        return {"authenticator": FreshdeskAuth(config["api_key"]), "config": config}
+
     def check_connection(self, logger: logging.Logger, config: Mapping[str, Any]) -> Tuple[bool, Optional[Any]]:
```

Comment:
For this, it looks like we want to use `CheckStream` - it uses AvailabilityStrategy under the hood, but has the benefit of making sure the stream you're checking is in the catalog (you don't need to instantiate it, just provide the name) and it will handle the try/excepting. I think the following should about do it: 
```
def check_connection(self, logger: logging.Logger, config: Mapping[str, Any]) -> Tuple[bool, Optional[Any]]:
    check_stream = CheckStream(stream_names=["settings"])
    return check_stream.check_connection(self, logger, config)
```

Reply:
Done. I removed `HTTPError` exception from try/except. But decided to leave as is the processing of other exceptions not related to HTTP